### DISCUSSION TITLE
Work on mali gpu

### DISCUSSION THREAD

**Discussion by YapoOriginal** [skyline-emu/skyline#1789] [Ideas]
I have a problem where the emulator stays black when opening any game, I think it's because of the processor ;v, they couldn't solve it, it's a Matepad Pro 12.6 inch help me pls

**Comment by s3lverd** [comment]
What games are you trying

**Reply by YapoOriginal** [reply]
Ori and the wild of The wisp

**Reply by s3lverd** [reply]
Mali doesn't work with very much, I don't even know if that game works on snapdragon either

**Reply by YapoOriginal** [reply]
but it is a very powerful mali (G78) and Ori is not a very demanding game, they should only make it compatible as with the Snapdragon, but that it does not work with any game is not normal, since I tried with other games and it still does not work, besides that It doesn't even start, it's not that it's laggy, because it tells me 0 fps and no loading screen appears, only the game controls with a totally black background.

**Reply by s3lverd** [reply]
Mali is just bad no matter how "good" you say it is

**Reply by guranu** [reply]
Mali is the worst gpu when it comes to emulation.

**Reply by YapoOriginal** [reply]
it literally has the same power as a snapdragon 845 graphic and you are telling me that it is not enough, or it is that you do not want to solve the problem which is not the performance of course

**Reply by YapoOriginal** [reply]
if it were as bad graphics as games can work exactly the same or with minimal differences between an Exynos and a Snapdragon, because if you insist that it is the power, I can only say that it is rather an unfounded favoritism

**Reply by ghost** [reply]
> if it were as bad graphics as games can work exactly the same or with minimal differences between an Exynos and a Snapdragon, because if you insist that it is the power, I can only say that it is rather an unfounded favoritism

You, my man clearly don't understand how emulation works. The comparison of adreno and mali gpus is exactly like comparison between a vega 11 and intel iris xe where vega demolishes intel in terms of drivers. It goes the same for adreno and mali. Next time ask samsung to get better drivers for their exynos chipsets. It requires much greater effort than on adreno to make games playable on mali. So you should be grateful you can at least play some games. Dolphin (official) and many other emulators don't even support mali properly. Also Ori isn't yet playable on mali devices.

**Reply by YapoOriginal** [reply]
Give me examples in android where the difference is very noticeable, since it is very evident, if the problem is the drivers, then as the creators of Games do, putting as an example of the simplest this Geshin Impact works the same in a Mali G78 than in a Adreno from Snapdragon 845, and it's true what you say I know very little about programming but if it can work with that with a much more basic game like Ori it can't, it's layers that emulate a game like Terraria and the same thing happens and you tell me what it is because of the drivers or because you don't know how to implement them correctly in the code?

**Reply by guranu** [reply]
try ps 2 emulation on mali with 6x resolution.

**Reply by hope3x** [reply]
![866661465724944385.png](https://user-images.githubusercontent.com/84674702/184529623-4f943989-3b3e-4e1c-a343-97fa41db4b85.png)

**Reply by hope3x** [reply]
Don't compare native games to emulation performance

**Reply by guranu** [reply]
> Give me examples in android where the difference is very noticeable, since it is very evident, if the problem is the drivers, then as the creators of Games do, putting as an example of the simplest this Geshin Impact works the same in a Mali G78 than in a Adreno from Snapdragon 845, and it's true what you say I know very little about programming but if it can work with that with a much more basic game like Ori it can't, it's layers that emulate a game like Terraria and the same thing happens and you tell me what it is because of the drivers or because you don't know how to implement them correctly in the code?

For fuck sake genshin impact was meant for mobile that's why it runs well. Games that you can only play on an emulator will run worse because they weren't meant to run on a phone.

**Reply by YapoOriginal** [reply]
I can play Wii games like super mario galaxy 2 with the Dolphin emulator almost perfectly, because they give it a few drops from time to time and you tell me that the problem is that it is Mali? and well I don't like Playstation 2 games but I'm going to try, I can also play pokemon Ultra sun in Citra at the highest possible quality 4X and it still works perfectly for me, if you want try it for yourselves.

**Reply by Dekuzearou** [reply]
A switch is way more heavier than a wii and 3ds you know

**Reply by guranu** [reply]
Exactly.

**Reply by YapoOriginal** [reply]
yes, I know that it is less powerful but I don't want it to work at 60 fps, only that it at least start, besides, I have already read messages from several of the programmers who work and they have said that they are compatibility problems, they have not said that it won't be able to work because of the difference in power, that's what I'm asking, not to be told that it's impossible, as if Mali wouldn't work at all knowing that there are compatible emulators and it also works very well, it's just making it compatible, and I can understand tell me that it is very difficult but to say that it is not possible is already a lie

**Reply by Dekuzearou** [reply]
Who told you it's impossible tho lol ofcourse games will work on a mali phone but just not as good as adreno phones

**Reply by YapoOriginal** [reply]
in the previous publication that I made they told me, besides that here in this publication they put that they are bad and do not work for that because of the drivers and others, if that is true they are indirectly saying that it is impossible, but that is not the only point I want the developers to listen to me and well try to at least give it support because I searched the internet and they implemented it, although I'm not sure because it was a youtube video I just want them to make it compatible with the Mali G78 graphics which is practically the latest generation, and Something that I forgot to say is that if there are errors in what is written, it is because I do not speak English and I am using the Google translator XD

**Reply by Dekuzearou** [reply]
Yes what they said was true but it's no impossible lol also devs are working hard on fixing mali GPUs so you should wait ig

**Reply by YapoOriginal** [reply]
thanks that was what I wanted to read, I'll wait then: v
